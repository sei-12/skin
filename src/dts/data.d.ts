
export namespace IData {
    export interface Bookmark {
        id: number
        title: string
        tags: string[]
        desc: string
        url: string
    }
}