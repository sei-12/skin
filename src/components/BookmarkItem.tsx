import { forwardRef } from "react"
import { IData } from "../data"
import { Card, CardContent, Stack, Typography } from "@mui/material"
import { globalColorTheme as GCT } from "../theme"

export type BookmarkItemProps = {
    data: IData.Bookmark

    onClickRemove: (key:string) => void
    onClickEdit: (key:string) => void
    focus: boolean
}

export const BookmarkItem = forwardRef<HTMLDivElement, BookmarkItemProps>(
	(props, ref) => {
		return (
            <div ref={ref}>
                <Card variant="elevation" sx={{ 
                    bgcolor: props.focus ? GCT.bookmarkItem.focusBg : GCT.bookmarkItem.bg,
                    borderRadius: 2.5,
                }}>
                    <CardContent>
                        <Typography variant="h5" sx={{
                            overflow: "hidden",
                            textOverflow: "ellipsis",
                            WebkitLineClamp: 1,
                            color: GCT.bookmarkItem.title,
                            fontWeight: "bold"
                        }}>{props.data.title}</Typography>
                        <Stack spacing={0.7} direction={"row"} sx={{overflow: "hidden"}}>
                            {props.data.tags.map((t, i) => <TagItem key={i} text={t}></TagItem>)}
                        </Stack>
                        <Typography
                            sx={{ 
                                color: GCT.bookmarkItem.desc,
                                fontWeight: 100,
                                overflow: "hidden",
                                display: "-webkit-box",
                                wordBreak: 'break-all' ,
                                lineClamp: 3, 
                                WebkitLineClamp: 3,
                                WebkitBoxOrient: "vertical",
                            }}
                        >{props.data.desc}</Typography>
                    </CardContent>
                </Card>
            </div>
		);
	}
);

function TagItem(props: {text: string}){
    return (
        <Typography
            sx={{
                color: GCT.bookmarkItem.tag,
                ":hover":{
                    color: GCT.bookmarkItem.tagHover
                }
            }}
        >#{props.text}</Typography>
    )
}
