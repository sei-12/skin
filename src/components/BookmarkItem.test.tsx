import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/vitest";
import { describe, it, expect, vi } from "vitest";
import { BookmarkItem, BookmarkItemProps } from "./BookmarkItem";
import { IData } from "../dts/data";

// Mock globalColorTheme
vi.mock("../lib/theme", () => ({
    globalColorTheme: {
        bookmarkItem: {
            focusBg: "#f0f0f0",
            bg: "#ffffff",
            title: "#000000",
            desc: "#666666",
            tag: "#888888",
        },
    },
}));

describe("BookmarkItem", () => {
    const mockData: IData.Bookmark = {
        id: 1,
        url: "aaa",
        title: "Sample Bookmark",
        desc: "This is a sample description for the bookmark item.",
        tags: ["tag1", "tag2", "tag3"],
    };

    const defaultProps: BookmarkItemProps = {
        data: mockData,
        onClickRemove: vi.fn(),
        onClickEdit: vi.fn(),
        focus: false,
    };

    it("renders the title, description, and tags", () => {
        render(<BookmarkItem {...defaultProps} />);

        // expect(screen.getByText("Sample Bookmark")).toBeInTheDocument();
        expect(screen.getByText("Sample Bookmark")).toBeVisible();
        expect(screen.getByText("This is a sample description for the bookmark item.")).toBeInTheDocument();
        expect(screen.getByText("#tag1")).toBeInTheDocument();
        expect(screen.getByText("#tag2")).toBeInTheDocument();
        expect(screen.getByText("#tag3")).toBeInTheDocument();
    });

    it("applies the correct background color when focused", () => {
        const { container, rerender } = render(<BookmarkItem {...defaultProps} focus={true} />);

        expect(container.querySelector(".MuiCard-root")).toHaveStyle("background-color: #f0f0f0");

        rerender(<BookmarkItem {...defaultProps} focus={false} />);

        expect(container.querySelector(".MuiCard-root")).toHaveStyle("background-color: #ffffff");
    });

    it("calls the correct functions on click actions", async () => {
        const user = userEvent.setup();
        const handleClickRemove = vi.fn();
        const handleClickEdit = vi.fn();

        render(
            <BookmarkItem
                {...defaultProps}
                onClickRemove={handleClickRemove}
                onClickEdit={handleClickEdit}
            />
        );


        // Simulate clicks
        // Replace these with actual buttons or clickable elements if present in your BookmarkItem
        
        await user.click(screen.getAllByRole('button')[0]); // どう書いたらいいかわからんかった
        await user.click(screen.getAllByRole('button')[1]);


        // Uncomment and replace with appropriate selectors if implemented
        expect(handleClickRemove).toHaveBeenCalledOnce();
        expect(handleClickEdit).toHaveBeenCalledOnce();
    });

    it("tag style", async () => {
        render(<BookmarkItem {...defaultProps} />);
        const tag = screen.getByText("#tag1");
        expect(tag).toHaveStyle("color: #888888");
    });
    
    it("snapshot",() => {
        const { container } = render(<BookmarkItem
               data={{
                title: "hello".repeat(100),
                url: "hello".repeat(2),
                desc: "aaa".repeat(200),
                id: 1,
                tags: Array(100).fill("tag")
               }}
               onClickEdit={() => {}}
               onClickRemove={() => {}}
               focus={false}
        ></BookmarkItem>)
        
        expect(container.querySelector(".MuiCard-root")).toMatchSnapshot()
    })
});