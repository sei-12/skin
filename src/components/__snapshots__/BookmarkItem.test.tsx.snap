// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookmarkItem > snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-wmg1xo-MuiPaper-root-MuiCard-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiCardContent-root css-13dhnfz-MuiCardContent-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1jxwjqx-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-grow css-2edihr-MuiGrid2-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-s1vonw-MuiTypography-root"
        >
          hellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohello
        </h5>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-auto css-1djy14n-MuiGrid2-root"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            aria-haspopup="true"
            class="MuiBox-root css-0"
            data-testid="open-bookmark-button"
            id="basic-button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yy668g-MuiSvgIcon-root"
              data-testid="MoreVertIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStack-root css-1wcha3e-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eriqsw-MuiTypography-root"
      >
        #
        tag
      </p>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-16to7o8-MuiTypography-root"
    >
      created at
       
      2025/02/07
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1hb8dbm-MuiTypography-root"
    >
      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
    </p>
  </div>
</div>
`;
